horizomer/utils/tree.py

```diff
@@ -15,6 +15,75 @@
 from skbio import TreeNode
 
 
+def order_nodes(tree, increase=True):
+    """Restore ordering of nodes in one tree based on another tree.
+
+    Parameters
+    ----------
+    tree : skbio.TreeNode
+        tree to order
+    increase : bool, optional
+        order nodes in increasing (True) or decreasing (False) order
+
+    Returns
+    -------
+    skbio.TreeNode
+        resulting ordered tree
+    """
+    res = tree.copy()
+    for node in res.postorder():
+        if node.is_tip():
+            node.n = 1
+        else:
+            node.n = sum(x.n for x in node.children)
+    for node in res.postorder():
+        if not node.is_tip():
+            child2n = {x: x.n for x in node.children}
+            node.children = []
+            for child in sorted(child2n, key=child2n.get, reverse=increase):
+                node.append(child)
+    for node in res.postorder():
+        delattr(node, 'n')
+    return res
```

Comment:
I'm not sure about the speed, but here is another version (that is fairly efficient).
```
def _cache_ntips(tree):
    for n in tree.postorder(include_self=True):
        if n.is_tip():
            n._n_tips = 1
        else:
            n._n_tips = sum(c._n_tips for c in n.children)
    return tree

def ladderize(tree, ascending=True):
    """
    Sorts tree according to the size of the subtrees.
    Parameters
    ----------
    tree : skbio.TreeNode
       Input tree where leafs correspond to features.
    Returns
    -------
    skbio.TreeNode
       A tree whose tips are sorted according to subtree size.
    Examples
    --------
    >>> from skbio import TreeNode
    >>> tree = TreeNode.read([u'((a,b)c, ((g,h)e,f)d)r;'])
    >>> print(tree.ascii_art())
                        /-a
              /c-------|
             |          \-b
    -r-------|
             |                    /-g
             |          /e-------|
              \d-------|          \-h
                       |
                        \-f
    >>> sorted_tree = ladderize(tree)
    >>> print(sorted_tree.ascii_art())
                        /-a
              /c-------|
             |          \-b
    -r-------|
             |          /-f
              \d-------|
                       |          /-g
                        \e-------|
                                  \-h
    """
    sorted_tree = tree.copy()
    sorted_tree = _cache_ntips(tree)

    for n in sorted_tree.postorder(include_self=True):
        sizes = [k._n_tips for k in n.children]
        idx = np.argsort(sizes)
        if not ascending:
            idx = idx[::-1]
        n.children = [n.children[i] for i in idx]
    return sorted_tree
```

Reply:
@nujinuji Jamie's implementation was tested working. It is basically very similar to our implementation, except for that he used some tricks that are more efficient (e.g., numpy's `argsort` is more efficient than Python `sorted`). It is for your reference. If you like, you can benchmark the two solutions to get an idea of efficiency.